src/deepqmc/sampling.py

```diff
@@ -365,6 +365,38 @@ def propagate_all(self):
         self.restart()
 
 
+def sort_nucleus_indices(idxs, coords):
+    # this heuristic takes a batch of nuclear indices for placing electrons
+    # and sorts them such that the local electronic spin is minimized
+    bs, n_electrons = idxs.shape
+    idx_new = []
+    idx_map = torch.arange(n_electrons).repeat(bs, 1)
+    idx_i = torch.randint(0, n_electrons, (bs,))
+    # starting from a randomly chosen electron, the nuclear indices of
+    # subsequent electrons are picked to correspond to the closest nucleus
+    while True:
```

Comment:
Isn't this loop guaranteed to take `n_electrons` cycles?

Reply:
Yes, it is. The while loop prevented introducing an unnecessairy variable and I had the construction for the break condition anyways. I can swap it to `for _ in range(n_electrons)` though to be more explicit about that.

Reply:
It's better like that because it's then obvious on first sight how long the loop takes.